samples/basics/binary_literals.jakt

```diff
@@ -2,4 +2,6 @@ function main() {
     println("{}", 0b10000000)
     println("{}", 0b111)
     println("{}", 0b11111111111110101)
+    println("{}", 0b0101_0101_0101_0101)
+    println("{}", 0b0101_0101_0101_0101_)
```

Comment:
Leading or trailing underscore should not be valid.

Reply:
Leading are not valid, rust allows trailing. Any particular reasoning for not allowing them?

Reply:
This is not rust, and there's no point in making it valid, so just don't :)

Reply:
and neither should more than one `_` be valid, FWIW

Reply:
(am I the only one that prefers `123'456'789`?)

Reply:
> (am I the only one that prefers `123'456'789`?)

That's always looked weird to me, especially considering we separate words in variable names with an underscore too

Reply:
FWIW:
- Allows trailing underscore: Rust, Swift
- Allows subsequent underscores in the middle: Java, Rust, Swift
- Disallows trailing underscore: Python, Java, JavaScript, C#, Kotlin, Go, PHP, Ruby

Reply:
@linusg same argument goes for this-is-not-Python-or-JavaScript. I'd argue for the current version as it's simpler to implement and slightly more flexible. I see no downside to allowing subsequent underscores and some minor upsides (multiple groups: `1111_1111__2222_2222`).

Reply:
> same argument goes for this-is-not-Python-or-JavaScript

I only said that because you tried to justify this decision with "rust does it" 🤷

The issue with it is readability - `1______________2` isn't readable, so it's not needed.

> I'd argue for the current version as it's simpler to implement

Sorry, but that's a terrible argument for making language design decisions :)